File: javascript/7kyu/max diff - easy.js
// https://www.codewars.com/kata/588a3c3ef0fbc9c8e1000095

// You must implement a function that returns the difference between the largest and the smallest value in a given list / array (lst) received as the parameter.

const maxDiff = (list) => {
    if (list.length <= 1) {
      return 0;
    }
    
    let min = list[0];
    let max = list[0];
    
    for (let i = 1; i < list.length; i++) {
      const current = list[i];
      min = Math.min(min, current);
      max = Math.max(max, current);
    }
    
    return max - min;
  };

File: java/7kyu/max diff - easy.java
// https://www.codewars.com/kata/588a3c3ef0fbc9c8e1000095

// You must implement a function that returns the difference between the largest and the smallest value in a given list / array (lst) received as the parameter.

public class Kata {

  public static int maxDiff(int[] lst) {
  
    if(lst.length == 0) return 0;

    int min = lst[0], 
        max = lst[0];

    for(int i = 1; i < lst.length; i++)
      if(lst[i] < min) min = lst[i];
      else if(lst[i] > max) max = lst[i];

    return max - min;
  }
}